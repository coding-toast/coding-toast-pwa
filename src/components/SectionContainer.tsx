import * as React from 'react'
import { Container } from 'react-bootstrap'

export enum BackgroundShade {
  PRIMARY = '#121212',
  SECONDARY = '#121f20'
}

interface ContainerProps {
  backgroundShade: BackgroundShade
}

const SectionContainer: React.FC<ContainerProps> = ({ backgroundShade, children }) => {
  return (
    <section style={{ background: backgroundShade }}>
      <Container>{children}</Container>
    </section>
  )
}

export default SectionContainer
