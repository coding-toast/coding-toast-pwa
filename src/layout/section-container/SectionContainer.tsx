import * as React from 'react';
import { Container } from 'react-bootstrap';

export enum BackgroundShade {
  PRIMARY = '#121212',
  SECONDARY = '#121f20'
}

interface IContainerProps {
  backgroundShade: BackgroundShade;
  style?: React.CSSProperties;
}

const SectionContainer: React.FC<IContainerProps> = ({ backgroundShade, style, children }) => {
  return (
    <section style={{ background: backgroundShade, paddingTop: '4em', paddingBottom: '3em', position: 'relative' }}>
      <Container style={style}>{children}</Container>
    </section>
  );
};

export default SectionContainer;
