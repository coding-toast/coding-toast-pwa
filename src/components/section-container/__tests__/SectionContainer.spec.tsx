import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SectionContainer, { BackgroundShade } from '../SectionContainer'

describe('<SectionContainer />', () => {
  const nestedContent = (
    <div>
      <h1>Hello World!</h1>
    </div>
  )
  it('renders correctly with Primary background shade', () => {
    const component = shallow(<SectionContainer backgroundShade={BackgroundShade.PRIMARY}>{nestedContent}</SectionContainer>)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('renders correctly with Secondary background shade', () => {
    const component = shallow(<SectionContainer backgroundShade={BackgroundShade.SECONDARY}>{nestedContent}</SectionContainer>)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('renders correctly with no children', () => {
    const component = shallow(<SectionContainer backgroundShade={BackgroundShade.PRIMARY} />)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('renders correctly with style injected container', () => {
    const component = shallow(<SectionContainer backgroundShade={BackgroundShade.PRIMARY} style={{ height: '80vh' }} />)
    expect(toJson(component)).toMatchSnapshot()
  })
})
