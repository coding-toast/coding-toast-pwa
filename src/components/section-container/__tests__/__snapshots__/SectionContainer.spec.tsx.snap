// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionContainer /> renders correctly with Primary background shade 1`] = `
<section
  style={
    Object {
      "background": "#121212",
    }
  }
>
  <Container
    fluid={false}
  >
    <div>
      <h1>
        Hello World!
      </h1>
    </div>
  </Container>
</section>
`;

exports[`<SectionContainer /> renders correctly with Secondary background shade 1`] = `
<section
  style={
    Object {
      "background": "#121f20",
    }
  }
>
  <Container
    fluid={false}
  >
    <div>
      <h1>
        Hello World!
      </h1>
    </div>
  </Container>
</section>
`;

exports[`<SectionContainer /> renders correctly with no children 1`] = `
<section
  style={
    Object {
      "background": "#121212",
    }
  }
>
  <Container
    fluid={false}
  />
</section>
`;
